pkg/command/permission_service_test.go

```diff
@@ -0,0 +1,223 @@
+package command_test
+
+import (
+	is "github.com/arquillian/ike-prow-plugins/pkg/command"
+	"github.com/arquillian/ike-prow-plugins/pkg/github"
+	. "github.com/arquillian/ike-prow-plugins/pkg/internal/test"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"gopkg.in/h2non/gock.v1"
+)
+
+var _ = Describe("Permission service with permission checks features", func() {
+
+	Context("Getting and validating user's permissions", func() {
+
+		client := NewDefaultGitHubClient()
+		var user is.PermissionService
+
+		BeforeEach(func() {
+			gock.Off()
+			user = is.PermissionService{
+				Client: client,
+				User:   "user",
+				PRLoader: &github.PullRequestLazyLoader{
+					Client:    client,
+					RepoOwner: "owner",
+					RepoName:  "repo",
+					Number:    1,
+				},
+			}
+		})
+
+		It("tests 'that is admin' function that should not approve the user when the permission is read", func() {
+			// given
+			gock.New("https://api.github.com").
+				Get("/repos/owner/repo/collaborators/user/permission").
+				Reply(200).
+				BodyString("{\"permission\": \"read\"}")
+
+			// when
+			permissionStatus, err := user.Admin()
+
+			// then
+			Ω(err).ShouldNot(HaveOccurred())
+			verifyStatusWithPredefinedUser(permissionStatus, false, "admin")
+		})
+
+		It("tests 'that is admin' function that should approve the user when the permission is admin", func() {
+			// given
+			gock.New("https://api.github.com").
+				Get("/repos/owner/repo/collaborators/user/permission").
+				Reply(200).
+				BodyString("{\"permission\": \"admin\"}")
+
+			// when
+			permissionStatus, err := user.Admin()
+
+			// then
+			Ω(err).ShouldNot(HaveOccurred())
+			verifyStatusWithPredefinedUser(permissionStatus, true, "admin")
+		})
+
+		It("tests 'that is pr creator' function that should not approve the user that is not the PR creator", func() {
+			// given
+			gock.New("https://api.github.com").
+				Get("/repos/owner/repo/pulls/1").
+				Reply(200).
+				BodyString(`{"user": {"login": "creator"}}`)
+
+			// when
+			permissionStatus, err := user.PRCreator()
+
+			// then
+			Ω(err).ShouldNot(HaveOccurred())
+			verifyStatusWithPredefinedUser(permissionStatus, false, "pull request creator")
+		})
+
+		It("tests 'that is pr creator' function that should approve the user that is the PR creator", func() {
+			// given
+			gock.New("https://api.github.com").
+				Get("/repos/owner/repo/pulls/1").
+				Reply(200).
+				BodyString(`{"user": {"login": "user"}}`)
+
+			// when
+			permissionStatus, err := user.PRCreator()
+
+			// then
+			Ω(err).ShouldNot(HaveOccurred())
+			verifyStatusWithPredefinedUser(permissionStatus, true, "pull request creator")
```

Comment:
Would it be possible (if quick enough) to make it a custom assertion `Expect(....)` (probably with `SatisfyAll`)?

Reply:
I'll try my best

Reply:
will be part of https://github.com/arquillian/ike-prow-plugins/issues/105